include/roboteam_ai/analysis/play-utilities/PlayDecider.h

```diff
@@ -9,10 +9,26 @@
 #include "analysis/play-utilities/Play.h"
 
 namespace rtt::ai::analysis {
-class PlayDecider {
-    PlayDecider() = default;
-    Play decideBestPlay(world::World* world, world::Field* field, std::vector<Play> validPlays);
-};
+    class PlayDecider {
+    public:
+        PlayDecider() = default;
+
+        /**
+            * Decides which play to choose out of the valid plays. Selects the play that best fits the world and field state. If you want to
+            * do meta logic, like making the play selection more aggressive or risky, you should do it in this function
+            * @param world the current world state
+            * @param field the current field state
+            * @param validPlays a vector of plays which have been vetted by the PlayDecider and are deemed appropriate for this situaton
+            * @return the play that best fits the world, and field.
+            */
+        std::shared_ptr<Play> decideBestPlay(world::World *world, world::Field *field, std::vector<std::shared_ptr<Play>> validPlays);
```

Comment:
Why does this have to be a shared_ptr, it's all owned by the PlayChecker, unique_ptr ftw